jupyterhub/apihandlers/base.py

```diff
@@ -39,8 +39,12 @@ def check_referer(self):
 
         - allow unspecified host/referer (e.g. scripts)
         """
-        host = self.request.headers.get("Host")
-        referer = self.request.headers.get("Referer")
+        host_header = self.request.headers.get("Host")
+        # http spec allows for port at end of host header but this causes
+        # cross site scripting check to give false positive
+        host, port = host_header.split(":", maxsplit=1)
```

Comment:
This line will raise an exception if there is no port in the header value.

Reply:
I will check it.

Reply:
I look for a : in the host header now prior to try to split it.  According to how I read the spec there can only be 1 :  in the host section.
Host = "Host" ":" host [ ":" port ] ; Section 3.2.2